### DISCUSSION TITLE
How can I reduce the space between ggplot2 figures and subsequent text in revealjs slides?

### DISCUSSION THREAD

**Discussion by jtr13** [quarto-dev/quarto-cli#10888] [Q&A] [answered]
### Description

Inspecting figures in my reveal slides shows that the css controling the spacing is:
```
.quarto-figure {
    position: relative;
    margin-bottom: 1em;
}
```
I have tried many different options to change `margin-bottom` to 0 such as adding the following to my custom .scss file:

```
.reveal .slides .quarto-figure {
  position: relative;
  margin-bottom: 0em; !important
}
```

and nothing works.  (It does work when I manually change it in Chrome tools.) Any tips?

**Comment by cscheid** [comment]
> and nothing works. (It does work when I manually change it in Chrome tools.) Any tips?

That probably means there's some other rule that's taking over. Can you provide a reprex?

**Comment by jtr13** [comment]
In my efforts to produce a reprex I figured out that the problem was caused by `auto-stretch: false` interacting with other elements, likely figure sizing or out.width though I haven't been able to pinpoint exactly what combination results in additional vertical space after a plot. Clearly though something is going on with `auto-stretch: false` as it wraps the plots in multiple additional tags:

**Example with auto-stretch: false**

````
format: 
  revealjs:
    auto-stretch: false
---

## auto-stretch: false

```{r}
library(ggplot2)
ggplot(mtcars, aes(wt, mpg)) + geom_point() + theme_grey(16)
```

This is a line of text below the plot.
````

<img width="400" alt="image" src="https://github.com/user-attachments/assets/67f2699d-d1af-48f6-82ca-1b31db307a87">

**Example with auto-stretch: true**

````
---
format: 
  revealjs:
    auto-stretch: true
---

## auto-stretch: true

```{r}
library(ggplot2)
ggplot(mtcars, aes(wt, mpg)) + geom_point() + theme_grey(16)
```

This is a line of text below the plot.
````

<img width="400" alt="image" src="https://github.com/user-attachments/assets/4d4bcb3d-7df9-41d0-b365-57266c0810de">

So, in short, I will stop using `auto-stretch: false` to avoid this issue.

**Reply by mcanouil** [reply, marked as answer]
Using the inspect/developer mode show what is adding the margins.

![image](https://github.com/user-attachments/assets/06bf4a96-1dcc-43f1-9a5a-07507f6d4f2b)

You only have to overwrite that rule, either in a CSS file via the `css` key, via SCSS using `theme` or via `include-in-header`.

````qmd
---
format: 
  revealjs:
    auto-stretch: false
    include-in-header:
      - text: |
          <style>
            .reveal p > img {
              margin: 0;
            }
          </style>
---

## auto-stretch: true

```{r}
plot(1)
```

This is a line of text below the plot.
````

**Reply by jtr13** [reply]
This is really helpful, thanks a lot!
